lib/array_intersection.rb

```diff
@@ -1,4 +1,19 @@
+
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  #raise NotImplementedError, "Intersection not implemented"
+
+  lookup_hash = {}
+  output = []
+
+  list1.each do |num|
+    lookup_hash[num] = false
+  end
+
+  list2.each do |num|
+    if lookup_hash[num] == false
+      output << num
+    end
+  end
```

Comment:
Why not just do this?

```suggestion
  list1.each do |num|
    lookup_hash[num] = true
  end

  list2.each do |num|
    if lookup_hash[num]
      output << num
    end
  end
```